docs/references/message-execution-properties.mdx

```diff
@@ -2,67 +2,55 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
 
 # Properties of message executions on ICP
 
-<MarkdownChipRow labels={["Reference"]} />
+<MarkdownChipRow labels={["Intermediate", "Reference", "Security"]} />
 
-There are a few key properties that one needs to understand about the execution model on ICP. Let's walk through them.
+## Asynchronous messaging model
 
-### Property 1
+ICP relies on an asynchronous messaging model. Compared to synchronous messaging like on Ethereum, this provides performance advantages because individual messages can be flexibly scheduled. However, asynchrounous message execution can also lead to sometimes unexpected or unintuitive behavior. Therefore, it is important to understand the properties of message execution. Potential security issues that arise in this model, such as reentrancy bugs, are discussed in the [security best practices on inter-canister calls](/docs/current/developer-docs/security/security-best-practices/inter-canister-calls).  
```

Comment:
```suggestion
ICP relies on an asynchronous messaging model. Compared to synchronous messaging like on Ethereum, this provides performance advantages because individual messages can be executed in parallel. However, asynchrounous message execution can also lead to sometimes unexpected or unintuitive behavior. Therefore, it is important to understand the properties of message execution. Potential security issues that arise in this model, such as reentrancy bugs, are discussed in the [security best practices on inter-canister calls](/docs/current/developer-docs/security/security-best-practices/inter-canister-calls).  
```